### ISSUE TITLE
Bulk carry forward can be done from any catalog #

### ISSUE THREAD

**Issue by emenslin** [specify/specify7#5130]
**Describe the bug**
Bulk carry forward will only add the records to the end of the catalog numbers but it can be done from any number to start with. This means that you could try to bulk carry from catalog number 1 and it would jump to the end to create all the records there and then the record set would have only a few records in it but the batch number would include more. 

**To Reproduce**

1. Go to any record in a db thats not the last one
2. Bulk carry forward a few records 
3. See the records are created with a gap in the numbers 

**Expected behavior**
With how the feature is currently implimented you should not be able to bulk carry forward unless it is the last record in the db. I think being able to bulk carry forward from any number could just be confusing to a user. Or at the very least there should be some sort of warning that numbers will be skipped 

**Screenshots**

https://github.com/user-attachments/assets/04dd4feb-5e25-4956-b6c5-b20d7106f1f5

Please, also fill out the following information manually:

- OS: Windows 11
- Browser: Chrome
- Specify 7 Version: edge
- Database Name: lsumzmammals9823
- Collection name: mammals
- User Name: spadmin

**Comment by emenslin** [comment]
Can recreate in edge (7.9.6.2)

**Comment by grantfitzsimmons** [comment]
@CarolineDenis I agree with @emenslin, this is confusing, but not something any user has reported. I think it is exceedingly uncommon to bulk carry forward with a count, and from what I understand, only the range has been used
